core/src/main/java/org/springframework/security/authentication/dao/DaoAuthenticationProvider.java

```diff
@@ -61,7 +61,11 @@ public class DaoAuthenticationProvider extends AbstractUserDetailsAuthentication
 	private UserDetailsPasswordService userDetailsPasswordService;
 
 	public DaoAuthenticationProvider() {
-		setPasswordEncoder(PasswordEncoderFactories.createDelegatingPasswordEncoder());
+		this(PasswordEncoderFactories.createDelegatingPasswordEncoder());
+	}
+
```

Comment:
I think that it would be great to add a little bit of javadoc here, with a little description and the `@since 5.8.3` tag